yt_dlp/extractor/sportdeutschland.py

```diff
@@ -1,95 +1,110 @@
+
         if len(videos) > 1:
-            playlist_id = parse_qs(url).get('playlistId', [None])[0]
-            if not self._yes_playlist(playlist_id, asset_id):
-                videos = [videos[int(playlist_id)]]
-
-            def entries():
-                for i, video in enumerate(videos, 1):
-                    video_id = video.get('uuid')
-                    video_url = video.get('url')
-                    if not (video_id and video_url):
-                        continue
-                    formats = self._extract_m3u8_formats(
-                        video_url.replace('.smil', '.m3u8'), video_id, 'mp4', fatal=False)
-                    if not formats and not self.get_param('ignore_no_formats'):
-                        continue
-                    yield {
-                        'id': video_id,
-                        'formats': formats,
-                        'title': title + ' - ' + (video.get('label') or 'Teil %d' % i),
-                        'duration': float_or_none(video.get('duration')),
-                    }
             info.update({
                 '_type': 'multi_video',
-                'entries': entries(),
-            })
-        else:
-            formats = self._extract_m3u8_formats(
-                videos[0]['url'].replace('.smil', '.m3u8'), asset_id, 'mp4')
-            section_title = strip_or_none(try_get(data, lambda x: x['section']['title']))
-            info.update({
-                'formats': formats,
-                'display_id': asset.get('permalink'),
-                'thumbnail': try_get(asset, lambda x: x['images'][0]),
-                'categories': [section_title] if section_title else None,
-                'view_count': int_or_none(asset.get('views')),
-                'is_live': asset.get('is_live') is True,
-                'timestamp': parse_iso8601(asset.get('date') or asset.get('published_at')),
-            })
+                'entries': self.processVideoOrStream(asset_id, video)
+            } for video in enumerate(videos) if video.get('formats'))
+
+        elif len(videos) == 1:
+            info.update(
+                self.processVideoOrStream(asset_id, videos[0])
+            )
+
+        livestream = meta.get('livestream')
+
+        if livestream is not None:
+            info.update(
+                self.processVideoOrStream(asset_id, livestream)
+            )
+
         return info
+
+    def processVideoOrStream(self, asset_id, video):
+        video_id = video.get('id')
+        video_src = video.get('src')
+        video_type = video.get('type')
+
+        token_data = self._download_json(
+            'https://api.sportdeutschland.tv/api/frontend/asset-token/' + asset_id
+            + '?type=' + video_type
+            + '&playback_id=' + video_src,
+            video_id
+        )
+
+        m3u8_url = "https://stream.mux.com/" + video_src + '.m3u8?token=' + token_data.get('token')
+        formats = self._extract_m3u8_formats(m3u8_url, video_id)
```

Comment:
```suggestion
        token = self._download_json(
            f'https://api.sportdeutschland.tv/api/frontend/asset-token/{asset_id}',
            video_id, query={'type': video_type, 'playback_id': video_src})['token']
        formats = self._extract_m3u8_formats('https://stream.mux.com/{video_src}.m3u8?token={token}', video_id)
```

Reply:
assuming I understood the CONTRIBUTING.md right with this assumption:

> Because, as I understand it, the [CONTRIBUTING.md ](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#mandatory-and-optional-metafields)states that info dicts should be accessed this way and not using for example video['id'].

shouldn't this be changed to
```suggestion
        token = self._download_json(
            f'https://api.sportdeutschland.tv/api/frontend/asset-token/{asset_id}',
            video_id, query={'type': video_type, 'playback_id': video_src}).get('token')
        formats = self._extract_m3u8_formats(f'https://stream.mux.com/{video_src}.m3u8?token={token}', video_id)
```
?

Reply:
That depends on if the token is required or not. If the token is required and extraction breaks without it, there is no point in sending the request in the first place. If the extractor has any fallbacks to extract formats, the token or similarly, only then does it make sense to not fatal.